src/scvi/module/_vae.py

```diff
@@ -372,17 +382,17 @@ def _regular_inference(
         if self.batch_representation == "embedding" and self.encode_covariates:
             batch_rep = self.compute_embedding(REGISTRY_KEYS.BATCH_KEY, batch_index)
             encoder_input = torch.cat([encoder_input, batch_rep], dim=-1)
-            qz, z = self.z_encoder(encoder_input, *categorical_input)
+            qz, z = self.z_encoder(encoder_input, cont_covs, *categorical_input)
```

Comment:
I think here batch_rep should be also concat to cont_covs to be treated as continuous covariate no?

Reply:
I would prefer to keep all seperated. So you could do deep injection of the batch embedding or covariate.

Reply:
That also sounds good as loong as all can be injected